project/apps/web/app/questions/components/question-table/QuestionTable.tsx

```diff
@@ -0,0 +1,154 @@
+"use client";
+
+import { QUERY_KEYS } from "@/constants/queryKeys";
+import { fetchQuestions } from "@/lib/api/question";
+import {
+  QuestionFiltersDto,
+  QuestionCollectionDto,
+  SortQuestionsQueryDto,
+} from "@repo/dtos/questions";
+import { useSuspenseQuery } from "@tanstack/react-query";
+import {
+  ControlledTableStateProps,
+  DataTable,
+} from "@/components/data-table/DataTable";
+
+import { columns } from "./columns";
+import { useQuestionsState } from "@/contexts/QuestionsStateContext";
+import { QuestionTableToolbar } from "./QuestionTableToolbar";
+import {
+  ColumnFiltersState,
+  PaginationState,
+  SortingState,
+  Updater,
+} from "@tanstack/react-table";
+import { startTransition, useState } from "react";
+import {
+  CATEGORY,
+  COMPLEXITY,
+} from "@repo/dtos/generated/enums/questions.enums";
+import useDebounce from "@/hooks/useDebounce";
+
+export function QuestionTable() {
+  const { confirmLoading, setConfirmLoading } = useQuestionsState();
+
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: 10,
+  });
+
+  const resetPagination = () => {
+    setPagination((prev) => ({
+      ...prev,
+      pageIndex: 0,
+    }));
+  };
+
+  const [sorting, setSorting] = useState<SortingState>([
+    {
+      id: "q_title",
+      desc: false,
+    },
+  ]);
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const debouncedColumnFilters = useDebounce(
+    columnFilters,
+    100,
+    () => setConfirmLoading(true),
+    () => setConfirmLoading(false),
+  );
+
+  const { data } = useSuspenseQuery<QuestionCollectionDto>({
+    queryKey: [
+      QUERY_KEYS.Question,
+      pagination,
+      sorting,
+      debouncedColumnFilters,
+    ],
+    queryFn: async () => {
+      setConfirmLoading(true);
```

Comment:
query triggers `confirmLoading` state to be `true`.

Reply:
Wait actually can we do, or is this not as good

1. utilize `useSuspenseQuery` to trigger the `Suspense` fallback on initial render to display the skeleton
2. utilize isPending for loading states -> this is applicable for when we are performing some kind of page functions, sorting functions, filtering functions

Reply:
>utilize isPending for loading states -> this is applicable for when we are performing some kind of page functions, sorting functions, filtering functions

correct me if im wrong, isn't this what `confirmLoading` state is doing currently? at the moment`confirmLoading` state is only set to true for create, update and delete operations. Initial render that displays the skeleton doesn't depend on this `confirmLoading` state.
Hence decided to piggy back this state for PSF operations